pkg/agent/openflow/network_policy.go

```diff
@@ -616,26 +625,57 @@ func getServiceMatchType(protocol *v1beta2.Protocol, ipv4Enabled, ipv6Enabled bo
 	return matchKeys
 }
 
-func (c *clause) generateServicePortConjMatches(port v1beta2.Service, priority *uint16, ipv4Enabled, ipv6Enabled bool) []*conjunctiveMatch {
-	matchKeys := getServiceMatchType(port.Protocol, ipv4Enabled, ipv6Enabled)
-	// Match all ports with the given protocol type if the matchValue is not specified (value is 0).
-	matchValue := uint16(0)
-	if port.Port != nil {
-		matchValue = uint16(port.Port.IntVal)
-	}
+func (c *clause) generateServicePortConjMatches(service v1beta2.Service, priority *uint16, ipv4Enabled, ipv6Enabled bool) []*conjunctiveMatch {
+	matchKeys := getServiceMatchType(service.Protocol, ipv4Enabled, ipv6Enabled)
+	ovsBitRanges := c.serviceToBitRanges(service)
 	var matches []*conjunctiveMatch
 	for _, matchKey := range matchKeys {
-		matches = append(matches,
-			&conjunctiveMatch{
-				tableID:    c.ruleTable.GetID(),
-				matchKey:   matchKey,
-				matchValue: matchValue,
-				priority:   priority,
-			})
+		for _, ovsBitRange := range ovsBitRanges {
+			matches = append(matches,
+				&conjunctiveMatch{
+					tableID:    c.ruleTable.GetID(),
+					matchKey:   matchKey,
+					matchValue: ovsBitRange,
+					priority:   priority,
+				})
+		}
 	}
 	return matches
 }
 
+func (c *clause) serviceToBitRanges(service v1beta2.Service) []types.BitRange {
+	var ovsBitRanges []types.BitRange
+	// If `EndPort` is equal to `Port`, then treat it as single port case.
+	if service.EndPort != nil && *service.EndPort > service.Port.IntVal {
+		// Add several antrea range services based on a port range.
+		portRange := thirdpartynp.PortRange{Start: uint16(service.Port.IntVal), End: uint16(*service.EndPort)}
+		bitRanges, err := portRange.BitwiseMatch()
+		if err != nil {
+			klog.Errorf("error when get bitMatch: %s", err.Error())
```

Comment:
```suggestion
			klog.Errorf("Error when getting BitRanges from %v: %v", portRange, err)
```
So that we can know what's the illegal input if it really happens.

Reply:
Done.